[c10d][Log] Use pg_id instead of pg_name for logging prefix

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132915
* __->__ #132058

When checking the logs of c10d, I found it showed that "[PG 7 rank 7]" which it actually means "[PG 1 rank 7]". So we need to use pg_id(aka, uid_) rather than pg_name_ because when creating subpgs, currently we need to call it multiple times, so this makes PG names are based on bumped up numbers (e.g, 7 rather than 1). Using pg_id is more accurate and consistent with other logging tools.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

fduwjj committed:
Update

[ghstack-poisoned]

wconstab reviewed (commented):
nice.

One favor to ask- can you help increase confidence that this is correct by writing up a little user-doc about what pg_name and uid_ are supposed to represent and why we have both?

fduwjj commented:
@wconstab sure, that also fall into the category of BE work, lol.

fduwjj committed:
Update on "[c10d][Log] Use pg_id instead of pg_name for logging prefix"


When checking the logs of c10d, I found it showed that "[PG 7 rank 7]" which it actually means "[PG 1 rank 7]". So we need to use pg_id(aka, uid_) rather than pg_name_ because when creating subpgs, currently we need to call it multiple times, so this makes PG names are based on bumped up numbers (e.g, 7 rather than 1). Using pg_id is more accurate and consistent with other logging tools.

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

shengbao-zheng commented:
LGTM!

fduwjj committed:
Update on "[c10d][Log] Use pg_id instead of pg_name for logging prefix"


When checking the logs of c10d, I found it showed that "[PG 7 rank 7]" which it actually means "[PG 1 rank 7]". So we need to use pg_id(aka, uid_) rather than pg_name_ because when creating subpgs, currently we need to call it multiple times, so this makes PG names are based on bumped up numbers (e.g, 7 rather than 1). Using pg_id is more accurate and consistent with other logging tools.

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1577,11 +1577,34 @@ std::string ProcessGroupNCCL::getNCCLWatchdogDebugInfo() {
+// have initialized PG in python layer. Before using split_group, even if
+// we are creating a new sub-PG, all ranks have to call the API at the same
+// time, and this makes `group_name` a unique identifier for a group (PG)
+// which makes it useful for debugging as well.
```
For comm_split, yes, I verified in the unit test, pg_name and pg_id are kind same now. Because to create a sub-pg across ranks, there is only one python call. So in the comm_split case, it's not technically `global_uid` unless it is hashed. But I agree, the name is quite confusing here. And for PG0 if it is default pg, the hash will be same across ranks. (I guess you mean sub-PG right, if rank 0 and 1 are not in the same subPG, then yes, the hash will be different).

wconstab replied:
> So in the comm_split case, it's not technically global_uid unless it is hashed.

is this intentional, or should we actually make it unique even for the comm split case?

fduwjj replied:
Good question, I don't know, maybe it's expected because now each rank only call the API once or fewer times, so there is no way for one to know how many more subPGs other ranks are going create.

fduwjj committed:
Update on "[c10d][Log] Use pg_id instead of pg_name for logging prefix"


When checking the logs of c10d, I found it showed that "[PG 7 rank 7]" which it actually means "[PG 1 rank 7]". So we need to use pg_id(aka, uid_) rather than pg_name_ because when creating subpgs, currently we need to call it multiple times, so this makes PG names are based on bumped up numbers (e.g, 7 rather than 1). Using pg_id is more accurate and consistent with other logging tools.

cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented:
Will send a renaming PR on top of this one.

fduwjj commented:
@pytorchbot merge
